example/lib/map_recorder_example.dart

```diff
@@ -0,0 +1,271 @@
+          ),
+          Container(
+            padding: const EdgeInsets.all(16),
+            color: Colors.white,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                Text(
+                  'Status: ${isRecording ? "Recording..." : isReplaying ? "Replaying..." : "Idle"}',
+                  style: Theme.of(context).textTheme.titleMedium,
+                  textAlign: TextAlign.center,
+                ),
+                const SizedBox(height: 8),
+                Text(
+                  'Playback State: $playbackState',
+                  style: Theme.of(context).textTheme.bodySmall,
+                  textAlign: TextAlign.center,
+                ),
+                const SizedBox(height: 16),
+                Row(
+                  children: [
+                    Expanded(
+                      child: ElevatedButton.icon(
+                        onPressed: isRecording || isReplaying
+                            ? null
+                            : _startRecording,
+                        icon: const Icon(Icons.fiber_manual_record),
+                        label: const Text('Record'),
+                        style: ElevatedButton.styleFrom(
+                          backgroundColor: Colors.red,
+                          foregroundColor: Colors.white,
+                        ),
+                      ),
+                    ),
+                    const SizedBox(width: 8),
+                    Expanded(
+                      child: ElevatedButton.icon(
+                        onPressed:
+                            isRecording && !isReplaying ? _stopRecording : null,
+                        icon: const Icon(Icons.stop),
+                        label: const Text('Stop'),
+                      ),
+                    ),
+                  ],
+                ),
+                const SizedBox(height: 8),
+                Row(
+                  children: [
+                    Expanded(
+                      child: ElevatedButton.icon(
+                        onPressed: recordedSequence != null &&
+                                !isRecording &&
+                                !isReplaying
+                            ? _replayRecording
+                            : null,
+                        icon: const Icon(Icons.play_arrow),
+                        label: const Text('Replay 2x'),
+                        style: ElevatedButton.styleFrom(
+                          backgroundColor: Colors.green,
+                          foregroundColor: Colors.white,
+                        ),
+                      ),
+                    ),
+                    const SizedBox(width: 8),
+                    Expanded(
+                      child: ElevatedButton.icon(
+                        onPressed: isReplaying ? _togglePause : null,
+                        icon: const Icon(Icons.pause),
+                        label: const Text('Pause'),
```

Comment:
It would be nice to update this text and icon after the user selects Pause so that they know to click the button again to restart the recording.